### ISSUE TITLE
TypeError: Object.groupBy is not a function. (In 'Object.groupBy(inventory, ({ type }) => type)', 'Object.groupBy' is undefined)

### ISSUE THREAD

**Issue by whiter001** [oven-sh/bun#7855]
### What is the problem this feature would solve?

TypeError: Object.groupBy is not a function. (In 'Object.groupBy(inventory, ({ type }) => type)', 'Object.groupBy' is undefined)

### What is the feature you are proposing to solve the problem?

[https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Object/groupBy](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Object/groupBy)

**Comment by Electroid** [comment]
We will support this once JSC (Safari) supports it.

**Comment by zloirock** [comment]
IIRC it's supported from Bun 1.0.19.

**Comment by Electroid** [comment]
Yep, looks like we already support this.

```js
> Object.groupBy
[Function: groupBy]
> Bun.version
'1.0.20'
```

**Comment by thunfisch987** [comment]
hi, Object.groupBy is working but typescript says it doesn't exist :thinking:
`Property 'groupBy' does not exist on type 'ObjectConstructor'.ts(2339)`
any help?
@types/bun is installed
this is my tsconfig.json:
```json
{
  "compilerOptions": {
    "lib": [
      "ESNext"
    ],
    "module": "esnext",
    "target": "esnext",
    "moduleResolution": "bundler",
    "moduleDetection": "force",
    "allowImportingTsExtensions": true,
    "noEmit": true,
    "composite": true,
    "strict": true,
    "downlevelIteration": true,
    "skipLibCheck": true,
    "jsx": "react-jsx",
    "allowSyntheticDefaultImports": true,
    "forceConsistentCasingInFileNames": true,
    "allowJs": true,
    "baseUrl": "src",
    "rootDir": "src",
    "outDir": "dist",
    "declaration": true,
    "paths": {
      "#utils": [
        "./utils"
      ],
      "#utils/*": [
        "./utils/*"
      ],
      "#console": [
        "./utils/console"
      ],
      "#console/*": [
        "./utils/console/*"
      ],
      "#puppet": [
        "./utils/puppet"
      ],
      "#puppet/*": [
        "./utils/puppet/*"
      ],
      "#types": [
        "./utils/types.ts"
      ]
    }
  },
  "exclude": [
    "prettier.config.mjs",
    "dist"
  ]
}
```

**Comment by thunfisch987** [comment]
or is that something that has to do with typescripts ESNext?
i could probably `declare` it but i don't know how exactly it would be implemented in typescript

**Comment by thunfisch987** [comment]
For anyone who is missing the typescript definition for Object.groupBy, you have to install `typescript@next`, because it's implemented in 5.4 which isn't out yet
